pkg/ruler/rulestore/bucketclient/bucket_client.go

```diff
@@ -177,17 +186,29 @@ func (b *BucketRuleStore) LoadRuleGroups(ctx context.Context, groupsToLoad map[s
 			for gr := range ch {
 				user, namespace, group := gr.GetUser(), gr.GetNamespace(), gr.GetName()
 				if user == "" || namespace == "" || group == "" {
-					return fmt.Errorf("invalid rule group: user=%q, namespace=%q, group=%q", user, namespace, group)
+					m.Lock()
+					e = fmt.Errorf("invalid rule group: user=%q, namespace=%q, group=%q", user, namespace, group)
```

Comment:
Is it better to use multi err rather than just wrapping errors?

Reply:
+1.. will do that